Extensions/timestamps.js

```diff
@@ -122,23 +121,58 @@ XKit.extensions.timestamps = new Object({
 		}
 
 		this.check_quota();
-		XKit.tools.add_css('#posts .post .post_content { padding-top: 0px; }', "timestamps");
-		if (this.preferences.posts.value || (this.preferences.inbox.value && XKit.interface.where().inbox)) {
-			XKit.post_listener.add("timestamps", this.add_timestamps);
-			this.add_timestamps();
-		}
-
-		if (this.preferences.reblogs.value !== "off") {
-			XKit.post_listener.add("timestamps", this.add_reblog_timestamps);
-			this.add_reblog_timestamps();
-		}
 
 		$(document).on("click", ".xkit-timestamp-failed-why", function() {
 			XKit.window.show("Timestamp loading failed.", "This might be caused by several reasons, such as the post being removed, becoming private, or the Tumblr server having a problem that it can't return the page required by XKit to load you the timestamp.", "error", "<div id=\"xkit-close-message\" class=\"xkit-button\">OK</div></div>");
 		});
 
-		if (this.preferences.only_on_hover.value) {
-			XKit.tools.add_css(" .xtimestamp {display: none; } .post:hover .xtimestamp {display: block; }", "timestamps_on_hover");
+		if (XKit.page.react) {
+			XKit.tools.async_add_function(async () => {
+				/* globals tumblr */
+				return await tumblr.getCssMap();
+			})
+			.then(({post, rightContent, reblog, reblogHeader, badge, blogLink, private}) => {
+				this.posts_class = post.map(x => `.${x}`).join(", ");
+				this.meatballs_class = rightContent.map(x => `.${x}`).join(", ");
+				this.reblogs_class = reblog.map(x => `.${x}`).join(", ");
+				this.reblog_headers_class = reblogHeader.map(x => `.${x}`).join(", ");
+				this.reblog_op_class = badge[1];
+				this.blog_link_class = blogLink.map(x => `.${x}`).join(", ");
+				this.private_posts_class = private.map(x => `.${x}`).join(", ");
```

Comment:
1. make sure you're using `XKit.css_map.getCssMap()` first; otherwise, the other functions will always return undefined
2. `keyToCss` takes a string argument, while `blogLink` here is an array from the CSS map, which we don't want because we then have to transform it into a selector list.

```js
XKit.css_map.getCssMap().then(() => {
	this.blog_link_selector = XKit.css_map.keyToCss("blogLink");
	// this.blog_link_selector is now ".hVK3L, ._23-Al, ._1PyOv, ._2A1wD"
});
```

Reply:
Ahhh there we go, I knew I should be giving it strings of the keys but I didn't know I needed to remove the curly braces and the keys in there. Thanks!

Reply:
that's not it, although the argument destructuring does become unnecessary.
your example used `XKit.css_map.keyToCss(blogLink)`, but `blogLink` is a variable name which points to an array. `XKit.css_map.keyToCss("blogLink")` is passing a string in directly

Reply:
Way too scatterbrained for programming, I'm unable to reproduce this error as well XD